<a href="https://colab.research.google.com/github/dinsor17/KKU_Data_mining/blob/master/visitors_k.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##อันดับแรกเลือก packages

การเรียกใช้ package ที่ชื่อว่า pandas

In [384]:
import pandas

##เชื่อมต่อกับ Google drive
หรือนำไฟล์จาก Drive มาที่ google colab

In [385]:
from google.colab import drive
drive.mount('/content/drive')

##ตั้งชื่อลิ้งค์
path = '/content/drive/My Drive/DWDM_material'

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


##การเรียกใช้ package ที่ชื่อว่า os

In [386]:
import os

dnp_attraction = os.path.join(path,'dnp_attraction.csv')
visitors = os.path.join(path,'visitors.csv')

##จากนั้นใช้ pandas โหลดข้อมูลมา

In [387]:
dnp_attraction_df = pandas.read_csv(dnp_attraction)
dnp_attraction_df.head()

Type-TH           Type-EN  ...         District   Subdistrict
0  สวนพฤกษศาสตร์  Botanical Garden  ...       วังน้ำเย็น  ทุ่งมหาเจริญ
1  สวนพฤกษศาสตร์  Botanical Garden  ...  เฉลิมพระเกียรติ          พุแค
2  สวนพฤกษศาสตร์  Botanical Garden  ...       เมืองระยอง          แกลง
3  สวนพฤกษศาสตร์  Botanical Garden  ...        พนมสารคาม    เขาหินซ้อน
4  สวนพฤกษศาสตร์  Botanical Garden  ...            นาโยง         นาโยง

[5 rows x 17 columns]

In [388]:
visitors_df = pandas.read_csv(visitors)
visitors_df.head()

จังหวัด รหัสจังหวัด        2549  ...      2559      2560      2561
0    กรุงเทพมหานคร       th-bm  36172138.0  ...  59196331  63575737  65534281
1      สมุทรปราการ       th-sp    958352.0  ...   2900180   3218934   3373639
2          นนทบุรี       th-no    830593.0  ...   2090095   2514677   2664025
3         ปทุมธานี       th-pt    868449.0  ...   1896549   2014281   2098344
4  พระนครศรีอยุธยา       th-pa   3373929.0  ...   7216514   7631557   8349613

[5 rows x 15 columns]

##เช็คดูจำนวนแถวและคอลัมน์


In [389]:
dnp_attraction_df.shape

(1431, 17)

In [390]:
visitors_df.shape

(77, 15)

##Missing Values

วิธีที่ 1 ลบ Missing ออกไปเลย

In [391]:
dnp_attraction_df.isnull()

Type-TH  Type-EN  SubType-TH  ...  Province  District  Subdistrict
0       False    False       False  ...     False     False        False
1       False    False       False  ...     False     False        False
2       False    False       False  ...     False     False        False
3       False    False       False  ...     False     False        False
4       False    False       False  ...     False     False        False
...       ...      ...         ...  ...       ...       ...          ...
1426    False    False       False  ...     False     False        False
1427    False    False       False  ...     False     False        False
1428    False    False       False  ...     False     False        False
1429    False    False       False  ...     False     False        False
1430    False    False       False  ...     False     False        False

[1431 rows x 17 columns]

In [392]:
dnp_attraction_df.isnull().any()

Type-TH                 True
Type-EN                 True
SubType-TH              True
SubType-EN              True
Attraction-name TH      True
Attraction-name EN      True
Organization-Unit-TH    True
Organization-Unit-EN    True
Latitude                True
Longtiude               True
Open-Houre              True
Image-URL               True
Facility                True
Contact                 True
Province                True
District                True
Subdistrict             True
dtype: bool

In [393]:
dnp_attraction_df_nm = dnp_attraction_df.dropna()

In [394]:
dnp_attraction_df_nm.isnull().any().any()

False

In [395]:
print(dnp_attraction_df.shape)
print(dnp_attraction_df_nm.shape)

(1431, 17)
(16, 17)


In [396]:
visitors_df.isnull()

จังหวัด  รหัสจังหวัด   2549   2550  ...   2558   2559   2560   2561
0     False        False  False  False  ...  False  False  False  False
1     False        False  False  False  ...  False  False  False  False
2     False        False  False  False  ...  False  False  False  False
3     False        False  False  False  ...  False  False  False  False
4     False        False  False  False  ...  False  False  False  False
..      ...          ...    ...    ...  ...    ...    ...    ...    ...
72    False        False  False  False  ...  False  False  False  False
73    False        False  False  False  ...  False  False  False  False
74    False        False  False  False  ...  False  False  False  False
75    False        False  False  False  ...  False  False  False  False
76    False        False  False  False  ...  False  False  False  False

[77 rows x 15 columns]

In [397]:
visitors_df.isnull().any()

จังหวัด        False
รหัสจังหวัด    False
2549            True
2550            True
2551            True
2552            True
2553            True
2554           False
2555           False
2556           False
2557           False
2558           False
2559           False
2560           False
2561           False
dtype: bool

In [398]:
visitors_df_nm = visitors_df.dropna()

In [399]:
visitors_df_nm.isnull().any().any()

False

In [400]:
print(visitors_df.shape)
print(visitors_df_nm.shape)

(77, 15)
(76, 15)


##การตรวจสอบว่า การลบข้อมูล missing ออก
จากตาราง dnp_attraction และ visitors จะทำให้ข้อมูลลดลงกี่ %

In [401]:
size_before_drop = dnp_attraction_df.shape[0]
size_after_drop = dnp_attraction_df.dropna().shape[0]

print(f'size before drop = {size_before_drop} \nsize after drop = {size_after_drop} \
    \nwhich is {100*((size_before_drop-size_after_drop)/size_before_drop)}')

size before drop = 1431 
size after drop = 16     
which is 98.88190076869321


In [402]:
size_before_drop = visitors_df.shape[0]
size_after_drop = visitors_df.dropna().shape[0]

print(f'size before drop = {size_before_drop} \nsize after drop = {size_after_drop} \
    \nwhich is {100*((size_before_drop-size_after_drop)/size_before_drop)}')

size before drop = 77 
size after drop = 76     
which is 1.2987012987012987


วิธีที่ 2 เต็มค่าที่เหมาะสม

In [403]:
dnp_attraction_df[['SubType-TH']] = dnp_attraction_df[['SubType-TH']].fillna(0)
dnp_attraction_df[['SubType-EN']] = dnp_attraction_df[['SubType-EN']].fillna(0)
dnp_attraction_df[['Attraction-name EN']] = dnp_attraction_df[['Attraction-name EN']].fillna(0)
dnp_attraction_df[['Organization-Unit-EN']] = dnp_attraction_df[['Organization-Unit-EN']].fillna(0)
dnp_attraction_df[['Latitude']] = dnp_attraction_df[['Latitude']].fillna(0)
dnp_attraction_df[['Longtiude']] = dnp_attraction_df[['Longtiude']].fillna(0)
dnp_attraction_df[['Open-Houre']] = dnp_attraction_df[['Open-Houre']].fillna(0)
dnp_attraction_df[['Image-URL']] = dnp_attraction_df[['Image-URL']].fillna(0)
dnp_attraction_df[['Facility']] = dnp_attraction_df[['Facility']].fillna(0)
dnp_attraction_df[['Contact']] = dnp_attraction_df[['Contact']].fillna(0)
dnp_attraction_df[['Province']] = dnp_attraction_df[['Province']].fillna(0)
dnp_attraction_df[['District']] = dnp_attraction_df[['District']].fillna(0)
dnp_attraction_df[['Subdistrict']] = dnp_attraction_df[['Subdistrict']].fillna(0)

In [404]:
dnp_attraction_df.isnull().any()

Type-TH                  True
Type-EN                  True
SubType-TH              False
SubType-EN              False
Attraction-name TH       True
Attraction-name EN      False
Organization-Unit-TH     True
Organization-Unit-EN    False
Latitude                False
Longtiude               False
Open-Houre              False
Image-URL               False
Facility                False
Contact                 False
Province                False
District                False
Subdistrict             False
dtype: bool

In [405]:
visitors_df[['2549']] = visitors_df[['2549']].fillna(0)
visitors_df[['2550']] = visitors_df[['2550']].fillna(0)
visitors_df[['2551']] = visitors_df[['2551']].fillna(0)
visitors_df[['2552']] = visitors_df[['2552']].fillna(0)
visitors_df[['2553']] = visitors_df[['2553']].fillna(0)

In [406]:
visitors_df.isnull().any()

จังหวัด        False
รหัสจังหวัด    False
2549           False
2550           False
2551           False
2552           False
2553           False
2554           False
2555           False
2556           False
2557           False
2558           False
2559           False
2560           False
2561           False
dtype: bool

##merge two tables (.merge())

In [407]:
merged_table = dnp_attraction_df.merge(visitors_df,how='left',left_on='Province',right_on='จังหวัด')
merged_table

Type-TH           Type-EN  ...       2560       2561
0     สวนพฤกษศาสตร์  Botanical Garden  ...  2155142.0  2252606.0
1     สวนพฤกษศาสตร์  Botanical Garden  ...  4204992.0  4356276.0
2     สวนพฤกษศาสตร์  Botanical Garden  ...  7319948.0  7761662.0
3     สวนพฤกษศาสตร์  Botanical Garden  ...  3239899.0  3408573.0
4     สวนพฤกษศาสตร์  Botanical Garden  ...  1514092.0  1578906.0
...             ...               ...  ...        ...        ...
1426        ทางทะเล            marine  ...        NaN        NaN
1427        ทางทะเล            marine  ...        NaN        NaN
1428        ทางทะเล            marine  ...        NaN        NaN
1429        ทางทะเล            marine  ...        NaN        NaN
1430        ทางทะเล            marine  ...        NaN        NaN

[1431 rows x 32 columns]

##เพิ่ม Column 'SUM' ที่รวมจำนวนนักท่องเที่ยวในแต่ละจังหวัดตั้งแต่ปี2549-2561

In [456]:
# sum for rows
df = visitors_df = pandas.read_csv(visitors)
df['SUM'] = df['2549']+df['2550']+df['2551']+df['2552']+df['2553']+df['2554']+df['2555']+df['2556']+df['2557']+df['2558']+df['2559']+df['2560']+df['2561']
df.head()

จังหวัด รหัสจังหวัด        2549  ...      2560      2561          SUM
0    กรุงเทพมหานคร       th-bm  36172138.0  ...  63575737  65534281  612808844.0
1      สมุทรปราการ       th-sp    958352.0  ...   3218934   3373639   27103976.0
2          นนทบุรี       th-no    830593.0  ...   2514677   2664025   22464137.0
3         ปทุมธานี       th-pt    868449.0  ...   2014281   2098344   18302732.0
4  พระนครศรีอยุธยา       th-pa   3373929.0  ...   7631557   8349613   75010518.0

[5 rows x 16 columns]

ค่าเฉลี่ยจำนวนนักท่องเที่ยวในปี2549-2561 เท่ากับ 34,626,780

In [409]:
df.mean()

2549    1.792726e+06
2550    1.898702e+06
2551    1.930408e+06
2552    1.643016e+06
2553    2.058383e+06
2554    2.261278e+06
2555    2.584253e+06
2556    2.819642e+06
2557    2.950993e+06
2558    3.234714e+06
2559    3.475489e+06
2560    3.763939e+06
2561    3.935314e+06
SUM     3.462678e+07
dtype: float64

## หาค่าเฉลี่ยจำนวนนักท่องเที่ยวในปี 2561

In [410]:
v_table = visitors_df[['จังหวัด','2549','2550','2551','2552','2553','2554','2555','2556','2557','2558','2559','2560','2561']]
v_table

จังหวัด        2549        2550  ...      2559      2560      2561
0     กรุงเทพมหานคร  36172138.0  35953546.0  ...  59196331  63575737  65534281
1       สมุทรปราการ    958352.0   1262082.0  ...   2900180   3218934   3373639
2           นนทบุรี    830593.0   1079877.0  ...   2090095   2514677   2664025
3          ปทุมธานี    868449.0   1050674.0  ...   1896549   2014281   2098344
4   พระนครศรีอยุธยา   3373929.0   3784617.0  ...   7216514   7631557   8349613
..              ...         ...         ...  ...       ...       ...       ...
72             ตรัง    726373.0    858638.0  ...   1465811   1514092   1578906
73           พัทลุง    448000.0    501320.0  ...   1487188   1569333   1641841
74          ปัตตานี    160320.0    131998.0  ...    257143    272739    281988
75             ยะลา    282463.0    269368.0  ...    624879    657317    680018
76         นราธิวาส    336473.0    284073.0  ...    646020    665710    692057

[77 rows x 14 columns]

In [411]:
v_table_m = v_table['2561']
v_table_m

0     65534281
1      3373639
2      2664025
3      2098344
4      8349613
        ...   
72     1578906
73     1641841
74      281988
75      680018
76      692057
Name: 2561, Length: 77, dtype: int64

## จำนวนนักท่องเที่ยวเฉลี่ยในปี 2561 เท่ากับ 3,935,314.4415584416 คน

In [412]:
v_table_m.mean()

3935314.4415584416

##วิธีเช็ค Summary ของแต่ละคอลัมน์ (count, min, max, mean)

In [413]:
merged_table.describe()

2549          2550  ...          2560          2561
count  1.310000e+02  1.310000e+02  ...  1.310000e+02  1.310000e+02
mean   1.634992e+06  1.711585e+06  ...  3.659830e+06  3.830910e+06
std    1.569444e+06  1.559765e+06  ...  3.245927e+06  3.409037e+06
min    1.603200e+05  1.319980e+05  ...  2.727390e+05  2.819880e+05
25%    6.604800e+05  6.701545e+05  ...  1.006934e+06  1.074060e+06
50%    1.075715e+06  1.216586e+06  ...  3.125937e+06  3.243323e+06
75%    2.111204e+06  2.058790e+06  ...  4.740683e+06  4.868330e+06
max    7.236704e+06  7.882541e+06  ...  1.740316e+07  1.821154e+07

[8 rows x 13 columns]

##เพื่อหาดูว่าจังหวัดไหนบ้างที่มีสถานที่ท่องเที่ยวเป็นสวนพฤกษศาสตร์

In [414]:
new_table = merged_table[['จังหวัด','Type-TH','Attraction-name TH','Open-Houre']]
new_table

จังหวัด  ...                             Open-Houre
0        สระแก้ว  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)
1        สระบุรี  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)
2          ระยอง  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)
3     ฉะเชิงเทรา  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)
4           ตรัง  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)
...          ...  ...                                    ...
1426         NaN  ...                                      0
1427         NaN  ...                                      0
1428         NaN  ...                                      0
1429         NaN  ...                                      0
1430         NaN  ...      ช่วงมกราคม-เมษายน ,ตลอดทั้งปี ฯลฯ

[1431 rows x 4 columns]

In [415]:
selected_list = merged_table['Type-TH']=='สวนพฤกษศาสตร์'
selected_list

0        True
1        True
2        True
3        True
4        True
        ...  
1426    False
1427    False
1428    False
1429    False
1430    False
Name: Type-TH, Length: 1431, dtype: bool

In [416]:
new_table[selected_list].head()

จังหวัด  ...                             Open-Houre
0     สระแก้ว  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)
1     สระบุรี  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)
2       ระยอง  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)
3  ฉะเชิงเทรา  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)
4        ตรัง  ...  ทุกวัน (เวลาราชการ 8.30 น. -16.30 น.)

[5 rows x 4 columns]

##[PD] select data

In [458]:
new_table = merged_table[['จังหวัด','Type-TH','2561']]
new_table

จังหวัด        Type-TH       2561
0        สระแก้ว  สวนพฤกษศาสตร์  2252606.0
1        สระบุรี  สวนพฤกษศาสตร์  4356276.0
2          ระยอง  สวนพฤกษศาสตร์  7761662.0
3     ฉะเชิงเทรา  สวนพฤกษศาสตร์  3408573.0
4           ตรัง  สวนพฤกษศาสตร์  1578906.0
...          ...            ...        ...
1426         NaN        ทางทะเล        NaN
1427         NaN        ทางทะเล        NaN
1428         NaN        ทางทะเล        NaN
1429         NaN        ทางทะเล        NaN
1430         NaN        ทางทะเล        NaN

[1431 rows x 3 columns]

In [418]:
selected_list = new_table['จังหวัด']=='ขอนแก่น'
selected_list

0       False
1       False
2       False
3       False
4       False
        ...  
1426    False
1427    False
1428    False
1429    False
1430    False
Name: จังหวัด, Length: 1431, dtype: bool

In [419]:
new_table[selected_list].head()

จังหวัด       2561              Type-TH
39  ขอนแก่น  5302795.0          สวนรุกขชาติ
40  ขอนแก่น  5302795.0          สวนรุกขชาติ
71  ขอนแก่น  5302795.0  ศูนย์กีฏวิทยาป่าไม้

In [420]:
new_table[new_table['จังหวัด']=='ขอนแก่น'].dropna()['2561'].mean()

5302795.0

###เพื่อดูความสัมพันธ์ว่าในแต่ละจังหวัดมีสถานที่ท่องเที่ยวอะไรบ้าง

In [421]:
transacs = merged_table.groupby('จังหวัด')['Type-TH'].apply(list)
transacs

จังหวัด
กาญจนบุรี                                                 [ทางบก]
ขอนแก่น           [สวนรุกขชาติ, สวนรุกขชาติ, ศูนย์กีฏวิทยาป่าไม้]
จันทบุรี                                    [ศูนย์กีฏวิทยาป่าไม้]
ฉะเชิงเทรา                           [สวนพฤกษศาสตร์, สวนรุกขชาติ]
ชลบุรี                     [สวนรุกขชาติ, สถานีเพาะเลี้ยงสัตว์ป่า]
ชัยภูมิ         [สวนรุกขชาติ, สวนรุกขชาติ, สถานีเพาะเลี้ยงสัตว...
ตรัง                               [สวนพฤกษศาสตร์, สวนพฤกษศาสตร์]
นครพนม                                              [สวนรุกขชาติ]
นครสวรรค์                                           [สวนรุกขชาติ]
นราธิวาส                        [สวนพฤกษศาสตร์, สวนรวมพรรณไม้ป่า]
น่าน                                   [สวนรุกขชาติ, สวนรุกขชาติ]
ปัตตานี                                             [สวนรุกขชาติ]
พังงา           [สถานีเพาะเลี้ยงสัตว์ป่า, ทางบก, ทางบก, ทางบก,...
พัทลุง                                            [สวนพฤกษศาสตร์]
พิจิตร                                              [สวนรุกขชาติ]
พิ

##Apriori

In [422]:
!pip install apyori

In [423]:
from apyori import apriori

In [463]:
list(apriori(transacs, min_support = 0.07, min_confidence = 0.7)) 


[RelationRecord(items=frozenset({'สวนรุกขชาติ'}), support=0.7380952380952381, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'สวนรุกขชาติ'}), confidence=0.7380952380952381, lift=1.0)]),
 RelationRecord(items=frozenset({'ทางบก', 'สวนพฤกษศาสตร์วรรณคดี'}), support=0.07142857142857142, ordered_statistics=[OrderedStatistic(items_base=frozenset({'สวนพฤกษศาสตร์วรรณคดี'}), items_add=frozenset({'ทางบก'}), confidence=0.75, lift=2.8636363636363633)]),
 RelationRecord(items=frozenset({'ทางบก', 'สวนรุกขชาติ'}), support=0.19047619047619047, ordered_statistics=[OrderedStatistic(items_base=frozenset({'ทางบก'}), items_add=frozenset({'สวนรุกขชาติ'}), confidence=0.7272727272727272, lift=0.9853372434017593)]),
 RelationRecord(items=frozenset({'สวนรุกขชาติ', 'สวนพฤกษศาสตร์วรรณคดี'}), support=0.09523809523809523, ordered_statistics=[OrderedStatistic(items_base=frozenset({'สวนพฤกษศาสตร์วรรณคดี'}), items_add=frozenset({'สวนรุกขชาติ'}), confidence=1.0, lift=1.3548387096774193

สรุปนักท่องเที่ยวส่วนใหญ่มักจะไปที่ 
- ทางบก และ สวนพฤกษศาสตร์วรรณคดี  ด้วยเท่ากับ 75% โดยมีค่า lift=2.86
- สวนรวมพรรณไม้ป่า และ สวนรุกขชาติ   ด้วยเท่ากับ 75%  โดยมีค่า lift=1.02
- สวนพฤกษศาสตร์วรรณคดี ต่อด้วย ทางบกกับสวนรุกขชาติ ด้วยเท่ากับ 75%  โดยมีค่า lift=3.94 ซึ่งมากกว่าคู่อื่น


##นักท่องเที่ยวจังหวัดขอนแก่นไปเที่ยวที่ไหนบ้าง

In [425]:
KK_table = merged_table['จังหวัด']=='ขอนแก่น'
KK = merged_table[KK_table]
KK

Type-TH  ...       2561
39          สวนรุกขชาติ  ...  5302795.0
40          สวนรุกขชาติ  ...  5302795.0
71  ศูนย์กีฏวิทยาป่าไม้  ...  5302795.0

[3 rows x 32 columns]

In [428]:
transacs_KK = KK.groupby('จังหวัด')['Type-TH'].apply(list)
transacs_KK

จังหวัด
ขอนแก่น    [สวนรุกขชาติ, สวนรุกขชาติ, ศูนย์กีฏวิทยาป่าไม้]
Name: Type-TH, dtype: object

In [429]:
list(apriori(transacs_KK, min_support = 0.07 , min_confidence = 0.7))

[RelationRecord(items=frozenset({'ศูนย์กีฏวิทยาป่าไม้'}), support=1.0, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ศูนย์กีฏวิทยาป่าไม้'}), confidence=1.0, lift=1.0)]),
 RelationRecord(items=frozenset({'สวนรุกขชาติ'}), support=1.0, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'สวนรุกขชาติ'}), confidence=1.0, lift=1.0)]),
 RelationRecord(items=frozenset({'ศูนย์กีฏวิทยาป่าไม้', 'สวนรุกขชาติ'}), support=1.0, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'ศูนย์กีฏวิทยาป่าไม้', 'สวนรุกขชาติ'}), confidence=1.0, lift=1.0), OrderedStatistic(items_base=frozenset({'ศูนย์กีฏวิทยาป่าไม้'}), items_add=frozenset({'สวนรุกขชาติ'}), confidence=1.0, lift=1.0), OrderedStatistic(items_base=frozenset({'สวนรุกขชาติ'}), items_add=frozenset({'ศูนย์กีฏวิทยาป่าไม้'}), confidence=1.0, lift=1.0)])]

##นักท่องเที่ยวจังหวัดชลบุรีไปเที่ยวที่ไหนบ้าง

In [451]:
CB_table = merged_table['จังหวัด']=='ชลบุรี'
CB = merged_table[CB_table]
CB

Type-TH                     Type-EN  ...        2560        2561
22              สวนรุกขชาติ                   Arboretum  ...  17403161.0  18211539.0
73  สถานีเพาะเลี้ยงสัตว์ป่า  Wildlife  Breeding Station  ...  17403161.0  18211539.0

[2 rows x 32 columns]

In [452]:
transacs_CB = CB.groupby('จังหวัด')['Type-TH'].apply(list)
transacs_CB

จังหวัด
ชลบุรี    [สวนรุกขชาติ, สถานีเพาะเลี้ยงสัตว์ป่า]
Name: Type-TH, dtype: object

In [453]:
list(apriori(transacs_CB, min_support = 0.08, min_confidence = 0.1)) 

[RelationRecord(items=frozenset({'สถานีเพาะเลี้ยงสัตว์ป่า'}), support=1.0, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'สถานีเพาะเลี้ยงสัตว์ป่า'}), confidence=1.0, lift=1.0)]),
 RelationRecord(items=frozenset({'สวนรุกขชาติ'}), support=1.0, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'สวนรุกขชาติ'}), confidence=1.0, lift=1.0)]),
 RelationRecord(items=frozenset({'สวนรุกขชาติ', 'สถานีเพาะเลี้ยงสัตว์ป่า'}), support=1.0, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'สวนรุกขชาติ', 'สถานีเพาะเลี้ยงสัตว์ป่า'}), confidence=1.0, lift=1.0), OrderedStatistic(items_base=frozenset({'สถานีเพาะเลี้ยงสัตว์ป่า'}), items_add=frozenset({'สวนรุกขชาติ'}), confidence=1.0, lift=1.0), OrderedStatistic(items_base=frozenset({'สวนรุกขชาติ'}), items_add=frozenset({'สถานีเพาะเลี้ยงสัตว์ป่า'}), confidence=1.0, lift=1.0)])]